src/components/WorkPoolEditForm.vue

```diff
@@ -32,35 +41,53 @@
 <script lang="ts" setup>
   import { showToast } from '@prefecthq/prefect-design'
   import { useValidationObserver } from '@prefecthq/vue-compositions'
-  import { ref } from 'vue'
+  import { ref, computed } from 'vue'
   import { useRouter } from 'vue-router'
-  import { SubmitButton } from '@/components'
-  import { useWorkspaceApi, useWorkspaceRoutes } from '@/compositions'
+  import { SubmitButton, WorkPoolTypeSelect, SchemaFormFieldsWithValues } from '@/components'
+  import { useCan, useWorkspaceApi, useWorkspaceRoutes } from '@/compositions'
   import { localization } from '@/localization'
-  import { WorkPool } from '@/models'
+  import { WorkPool, WorkPoolEdit } from '@/models'
+  import { mapper } from '@/services/Mapper'
+  import { Schema } from '@/types/schemas'
 
   const props = defineProps<{
     workPool: WorkPool,
   }>()
 
   const api = useWorkspaceApi()
+  const can = useCan()
   const router = useRouter()
   const routes = useWorkspaceRoutes()
   const { validate, pending } = useValidationObserver()
 
   const description = ref<string | null | undefined>(props.workPool.description)
+  const type = ref<string>(props.workPool.type)
   const concurrencyLimit = ref<number | null | undefined>(props.workPool.concurrencyLimit)
+  const schema = computed<Schema>(() => mapper.map('SchemaResponse', props.workPool.baseJobTemplate.variables ?? {}, 'Schema'))
+  // Set parameters to the default values from the schema so they are pre-filled in the form
+  const parameters = ref<Record<string, unknown>>(
+    Object.entries(props.workPool.baseJobTemplate.variables?.properties ?? {}).reduce((acc, [key, value]) => ({ ...acc, [key]: value?.default }), {}),
+  )
+
+  const schemaHasProperties = computed(() => {
+    const { properties } = schema.value
+
+    return properties && Object.keys(properties).length > 0
+  })
 
   function cancel(): void {
     router.back()
   }
 
   const submit = async (): Promise<void> => {
+    const baseJobTemplateSchema = mapper.map('WorkerSchemaProperty', { values: parameters.value, schema: props.workPool.baseJobTemplate }, 'WorkerSchemaPropertyRequest')
```

Comment:
I'm a little surprised this isn't happening as part of the `updateWorkPool` call - I don't frequently see us invoke the mapper before we make an API request in a component

Reply:
I moved the mapper into the `updateWorkPool` call and updated `updateWorkPool` to accept the inputs from the form. Let me know if that pattern works better.